cylc/flow/cfgspec/suite.py

```diff
@@ -295,6 +291,15 @@ def upg(cfg, descr):
     u.obsolete('7.8.1', ['runtime', '__MANY__', 'events', 'reset timer'])
     u.obsolete('8.0.0', ['runtime', '__MANY__', 'job', 'shell'])
 
+    # TODO: not obvious how to do this using upgrader
```

Comment:
For now I'm happy with this *not* being done using the upgrader as this way it won't raise a deprecation notice so won't present a barrier to Cylc8 adoption.

At somepoint in Cylc8 we should deprecate the old syntax and remove support for it in Cylc9?

Reply:
I agree - deprecate then remove.

Reply:
Should we deprecate in Cylc8.0.0 or some later Cylc8.x.y?

Reply:
I vote sooner rather than later.

Reply:
I would rather we don't have mixed syntax, as, if nothing else that's just, yuck.

My only worry with deprecating it now is that we've been telling people there is no upgrade process for Cylc8, if they start their (perfectly valid Cylc7.8.x) suites under Cylc8 and get pages of deprecation warnings...

Reply:
Deprecation doesn't break anything, it's just warns users of the new better way of doing things.

Reply:
But anyway, have we really been telling users that no upgrade will be needed at all, for cylc-7 suite defns?  (I haven't).

Reply:
> have we really been telling users that no upgrade will be needed at all

Yup.

Reply:
Did we officially agree on that?  (Just asking, maybe we did ... I know we did discuss it at some point)

Reply:
I do recall that we agreed to obsolete all pre-cylc-7 syntax, at cylc 8.  So users who have been living with cylc-6 deprecation warnings will have to upgrade!

Reply:
But I don't recall that we agreed not to deprecate (as opposed to obsolete) any cylc-7 syntax, at cylc-8.

Reply:
I think we've agreed to deprecate for Cylc8, obsolete at Cylc9 and educate users pre-release.

Ideally if we could group the deprecation warnings to avoid users seeing one for each recurrence section...

Reply:
> Should cylc validate fail on mixed syntax?

Yes, seems sensible.